src/mainloop.rkt

```diff
@@ -78,7 +79,11 @@
   (debug #:from 'progress #:depth 3 "[1/2] Preparing points")
   ;; If the specification is given, it is used for sampling points
   (timeline-event! 'analyze)
-  (*sampler* (make-sampler (*output-repr*) precondition-prog (or specification prog)))
+  (define symmetry-groups (filter (lambda (group) (> (length group) 1)) (connected-components (or specification prog))))
+  (define preprocess-structs (append preprocess (map symmetry-group symmetry-groups)))
+  (*herbie-preprocess* preprocess-structs)
+  (*sampler* (make-sampler (*output-repr*) precondition-prog (list (or specification prog)) (*herbie-preprocess*)))
```

Comment:
Should we add a timeline event for symmetry groups? I think so!

Reply:
Yep!